Sources/CMakeLists.txt

```diff
@@ -49,74 +66,139 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     SET(DEBUG TRUE)
 endif()
 
-add_definitions(-D_REENTRANT=1)
-add_definitions(-D_MT=1)
+## ** RAKE! start compiler specific flags section **
+## ** RAKE! Borrowed from dhewm3 project, need to **
+## ** RAKE! clean up for SeriousEngine use. Also  **
+## ** RAKE! need to make this pandora safe.	  **
+# compiler specific flags
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+	# This section and the like are for flags/defines that can be shared between 
+	# c and c++ compile options
+	add_compile_options(-Wall)
+	add_compile_options(-pipe)
+	add_compile_options(-fPIC)
+	add_compile_options(-march=native)
+	add_definitions(-D_REENTRANT=1)
+	add_definitions(-D_MT=1)
+
+	## Add your custom C and CXX flags here.
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+	## For C flags
+	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS} -g -D_DEBUG=1 -DDEBUG=1 -O0")
```

Comment:
Ah ok, I will just have it use systems zlib, and/or have an option to decide on which one to use, but have it compile with c++.